build.gradle.kts

```diff
@@ -164,9 +164,12 @@ subprojects {
     }
 
     tasks.withType<KotlinCompile>().configureEach {
+        val hasSerializationPlugin = plugins.hasPlugin(libs.plugins.kotlinxSerialization.get().pluginId)
+
         compilerOptions {
             allWarningsAsErrors = true
             jvmTarget = maxKotlinJvmTarget
+            if (hasSerializationPlugin) optIn = listOf("kotlinx.serialization.ExperimentalSerializationApi")
```

Comment:
Could we just globally define this opt in, or would it fail in projects not depending on kotlinx.serialization?

Reply:
It would fail.